src/main/java/net/minecraftforge/common/SimpleAnvilRecipe.java

```diff
@@ -0,0 +1,142 @@
+package net.minecraftforge.common;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+import net.minecraft.block.Block;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.world.World;
+import net.minecraftforge.oredict.OreDictionary;
+
+public class SimpleAnvilRecipe implements IAnvilRecipe
+{
+    protected final List<ItemStack> left;
+    protected final List<ItemStack> right;
+    protected final ItemStack output;
+    protected final int expCost;
+    protected final int matCost;
+
+    public SimpleAnvilRecipe(ItemStack result, Object inputLeft, Object inputRight, int materialCost, int experienceCost)
+    {
+        this.output = result.copy();
+        this.left = processInput(inputLeft);
+        this.right = processInput(inputRight);
+        this.matCost = materialCost;
+        this.expCost = experienceCost;
+        if (this.left == null || (this.right == null && inputRight != null)) {
+            String ret = "Invalid simple anvil recipe: " + inputLeft + ", " + inputRight + ", " + this.output;
+            throw new RuntimeException(ret);
+        }
+    }
+
+    private List<ItemStack> processInput(Object input)
+    {
+        if (input instanceof String)
+        {
+            return OreDictionary.getOres((String)input);
+        }
+        else
+        {
+            ItemStack stack = null;
+            if (input instanceof ItemStack)
+            {
+                stack = ((ItemStack)input).copy();
+            }
+            else if (input instanceof Item)
+            {
+                stack = new ItemStack((Item)input, 1, OreDictionary.WILDCARD_VALUE);
+            }
+            else if (input instanceof Block)
+            {
+                stack = new ItemStack((Block)input, 1, OreDictionary.WILDCARD_VALUE);
+            }
+            return stack != null ? Collections.singletonList(stack) : OreDictionary.EMPTY_LIST;
+        }
+    }
+
+    @Override
+    public boolean matches(ItemStack inputLeft, ItemStack inputRight, String newName, World world)
+    {
+        return checkMatch(this.left, inputLeft, false) && checkMatch(this.right, inputRight, true);
+    }
+
+    private boolean checkMatch(List<ItemStack> target, ItemStack tableItem, boolean checksize)
+    {
+        if (target == OreDictionary.EMPTY_LIST)
+        {
+            return tableItem == null;
+        }
+        else if (checksize && tableItem.stackSize < this.matCost)
```

Comment:
Note: This line has a NPE if the material slot is empty.
Replace line with: `else if (checksize && (tableItem == null || tableItem.stackSize < this.matCost))`